### PR TITLE
Fix overlap vcf2diploid

### PR DESCRIPTION
**Pull request by yunfeiguo**
- [ ] [RES-XXXX](https://binatechnologies.atlassian.net/browse/RES-XXXX)
- [ ] Blocking PRs: ADD HERE IF ANY
* People
  - [ ] Reviewers (>=1): ADD HERE
  - [ ] Merger (>=1): ADD HERE
  - [ ] FYI: ADD HERE IF ANY
- [ ] Release note (if needed)
- [ ] No code copied from outside Bina 
- [ ] Tests (Unit, Workflow, ITL & Gherkin) ing
- [ ] PR labels, milestones & assignees

# Summary

check insertion/duplication overlapping with deleted bp.

### CODE REVIEW

File: src/main/java/com/bina/varsim/tools/simulation/VCF2diploid.java [hunk 1]
```diff
@@ -361,24 +361,33 @@ private boolean addVariant(final byte[] maskedSequence, final Sequence reference
...
-            // DELETED_BASE is used as a flag to mark that this position has been processed/modified
-            if (maskedSequence[p - 1] == DELETED_BASE
-                    // insertions may not be next to SNPs or deletions, this is to avoid problem of DUP/INV
-                    //why p - 1 for maskedSequence? is it 1-based or 0-based?
-                    || maskedSequence[p - 1] != referenceSequence.byteAt(p)) {
-                overlap = true;
+        if (referenceAlleleLength == 0) {//insertion or duplication
+            // insertions may not be surrounded by deletions
+                if (maskedSequence[position - 1] == DELETED_BASE &&
+                        (position >= 2 && maskedSequence[position - 2] == DELETED_BASE) &&
```
Comment (marghoob) [inline src/main/java/com/bina/varsim/tools/simulation/VCF2diploid.java]: Why are you looking at `position 2`?

File: src/main/java/com/bina/varsim/tools/simulation/VCF2diploid.java [hunk 2]
```diff
@@ -361,24 +361,33 @@ private boolean addVariant(final byte[] maskedSequence, final Sequence reference
...
+                    overlap = true;
+                }
             }
         }
 
         if (overlap) {
             try {
                 if (insertions != null) {
-                    log.warn("Variant overlap at " + referenceSequence.getName() + ":"
+                    log.warn("Variant (" + variant + ") overlap at " + referenceSequence.getName() + ":"
```
Comment (marghoob) [inline src/main/java/com/bina/varsim/tools/simulation/VCF2diploid.java]: 👍

File: src/main/java/com/bina/varsim/tools/simulation/VCF2diploid.java (hunk 1 above)
Comment (yunfeiguo) [reply src/main/java/com/bina/varsim/tools/simulation/VCF2diploid.java]: `position` is 1-based, while `maskedSequence` is 0-based. `position - 2` is the bp before insertion locus, `position - 1` is the bp after insertion locus. 

actually there is no need to check `position`, which is 2bp after insertion locus. I just removed it. Thanks.

**Comment by yunfeiguo** [comment]
@marghoob ok now?

**Comment by marghoob** [comment]
👍  Thanks.

**Comment by yunfeiguo** [comment]
Great. Thank you!

**Pull request merged**